src/centralized_telescope/params.rs

```diff
@@ -1,16 +1,147 @@
-//! ALBA's Setup structure
+//! Centralized Telescope's Params structure comprising the internal parameters
 
-#![doc = include_str!("../../docs/rustdoc/centralized_telescope/params.md")]
+use std::f64::consts::LOG2_E;
 
-/// Setup input parameters
+/// Internal parameters
 #[derive(Debug, Clone, Copy)]
 pub struct Params {
-    /// Soundness security parameter
-    pub soundness_param: f64,
-    /// Completeness security parameter
-    pub completeness_param: f64,
-    /// Approximate size of the prover set to lower bound
-    pub set_size: u64,
-    /// Lower bound to prove on prover set
-    pub lower_bound: u64,
+    /// Number of prover set's elements
+    pub proof_size: u64,
+    /// Maximum number of retries to find a proof
+    pub max_retries: u64,
+    /// Maximum number of subtrees to search to find a proof
+    pub search_width: u64,
+    /// Probability that a tuple of element is a valid proof
+    pub valid_proof_probability: f64,
+    /// Maximum number of DFS calls permitted to find a proof
+    pub dfs_bound: u64,
+}
+
+impl Params {
+    /// Setup algorithm taking a Params as input and returning Params parameters (u,d,q)
+    pub fn new(
+        soundness_param: f64,
+        completeness_param: f64,
+        set_size: u64,
+        lower_bound: u64,
+    ) -> Self {
+        let set_size_f64 = set_size as f64;
+        let lower_bound_f64 = lower_bound as f64;
+
+        let proof_size_f64 = {
+            let numerator = soundness_param + completeness_param.log2() + 5.0 - LOG2_E.log2();
```

Comment:
give more descriptive constant names to 5.0

Reply:
Done